src/main/java/com/ieltswise/service/impl/GoogleEventsServiceImpl.java

```diff
@@ -135,14 +146,22 @@ private String createUrl(String tutorId, ZonedDateTime startOfMonth, ZonedDateTi
         return apiUrl + "?timeMin=" + encodedTimeMin + "&timeMax=" + encodedTimeMax + "&key=" + googleCredentialKey;
     }
 
-    private List<FreeAndBusyHoursOfTheDay> findAllEventsByYearAndMonth(JSONArray jsonArray, ZonedDateTime startOfMonth,
-                                                                       ZonedDateTime endOfMonth) {
+
+    private List<FreeAndBusyHoursOfTheDay> findAllEventsByYearAndMonth(JSONArray jsonArray,
+                                                                       ZonedDateTime startOfMonth,
+                                                                       ZonedDateTime endOfMonth,
+                                                                       Map<DayOfWeek, List<TimeSlot>> schedule) {
 
```

Comment:
**Add Unit test methods for this method on various variants of the received data.**

Reply:
done